### PR TITLE
Refresh mapview after sync structures/task/events in the selected operational area

### PR DESCRIPTION
**Pull request by githengi**
PR for Issue [https://github.com/OpenSRP/opensrp-client-reveal/issues/50](https://github.com/OpenSRP/opensrp-client-reveal/issues/50)

Depends on [https://github.com/OpenSRP/opensrp-client-core/pull/149](https://github.com/OpenSRP/opensrp-client-core/pull/149)

### CODE REVIEW

File: opensrp-reveal/src/main/java/org/smartregister/reveal/sync/LocationTaskIntentService.java [hunk 1]
```diff
@@ -47,11 +44,33 @@ public void run() {
                 SyncServiceJob.scheduleJobImmediately(SyncServiceJob.TAG);
             }
         });
-        if (!syncedStructures.isEmpty() || !synchedTasks.isEmpty()) {
+
+        if (hasChangesInCurrentOperationalArea(syncedStructures, synchedTasks)) {
             Intent intent = new Intent(STRUCTURE_TASK_SYNCHED);
             LocalBroadcastManager.getInstance(getApplicationContext()).sendBroadcast(intent);
```
Comment (vincent-karuri) [inline opensrp-reveal/src/main/java/org/smartregister/reveal/sync/LocationTaskIntentService.java]: Does this serve a different purpose than the broadcast here https://github.com/OpenSRP/opensrp-client-reveal/blob/21b831b741147ceffcc300c156e5b5655944fda7/opensrp-reveal/src/main/java/org/smartregister/reveal/sync/RevealClientProcessor.java#L95?

If so, please explain.

Comment (githengi) [reply opensrp-reveal/src/main/java/org/smartregister/reveal/sync/LocationTaskIntentService.java]: Yes, Events and synced independenty from Tasks and Structures.

So we are broadcasting in 2 times
1.) When a new spray event is synced
2.) When a new structure or Task is synced

File: opensrp-reveal/src/main/java/org/smartregister/reveal/sync/LocationTaskIntentService.java [hunk 2]
```diff
@@ -47,11 +44,33 @@ public void run() {
...
 
     }
 
+    private boolean hasChangesInCurrentOperationalArea(List<Location> syncedStructures, List<Task> synchedTasks) {
+        Location operationalAreaLocation = Utils.getOperationalAreaLocation(PreferencesUtil.getInstance().getCurrentOperationalArea());
+        if (operationalAreaLocation == null)
+            return false;
+        if (syncedStructures != null) {
+            for (Location structure : syncedStructures) {
+                if (operationalAreaLocation.getId().equals(structure.getProperties().getParentId())) {
```
[the remark this answers is not in the document]
Comment (githengi) [reply opensrp-reveal/src/main/java/org/smartregister/reveal/sync/LocationTaskIntentService.java]: Okay :smile:

File: opensrp-reveal/src/main/java/org/smartregister/reveal/sync/LocationTaskIntentService.java (hunk 1 above)
Comment (vincent-karuri) [reply opensrp-reveal/src/main/java/org/smartregister/reveal/sync/LocationTaskIntentService.java]: Cool.

**Pull request merged**